v5/core/base_service.go

```diff
@@ -278,7 +278,7 @@ func (service *BaseService) buildUserAgent() string {
 // SetUserAgent sets the user agent value.
 func (service *BaseService) SetUserAgent(userAgentString string) {
 	if userAgentString == "" {
-		service.UserAgent = service.buildUserAgent()
+		userAgentString = service.buildUserAgent()
```

Comment:
This code coverage work uncovered one minor bug!!!

Reply:
That's a nice catch!